ddtrace/internal/sampling.py

```diff
@@ -208,44 +211,102 @@ def apply_span_sampling_tags(self, span):
 
 def get_span_sampling_rules():
     # type: () -> List[SpanSamplingRule]
-    json_rules_raw = os.getenv("DD_SPAN_SAMPLING_RULES")
-    if json_rules_raw is None:
+    json_rules = get_span_sampling_json()
+    validate_json(json_rules)
+    sampling_rules = []
+    for rule in json_rules:
+        # If sample_rate not specified default to 100%
+        sample_rate = rule.get("sample_rate", 1.0)
+        service = rule.get("service")
+        name = rule.get("name")
+        # If max_per_second not specified default to no limit
+        max_per_second = rule.get("max_per_second", -1)
+        if service:
+            _check_unsupported_pattern(service)
+        if name:
+            _check_unsupported_pattern(name)
+
+        try:
+            sampling_rule = SpanSamplingRule(
+                sample_rate=sample_rate, service=service, name=name, max_per_second=max_per_second
+            )
+        except Exception as e:
+            raise ValueError("Error creating single span sampling rule {}: {}".format(json.dumps(rule), e))
+        sampling_rules.append(sampling_rule)
+    return sampling_rules
+
+
+def validate_json(json_rules):
+    schema = {
+        "type": "array",
+        "anyOf": [
```

Comment:
I don't think you need this `anyOf`, since there is only one valid value... an array of objects.


```python
   schema = {
        "type": "array",
        "items": {
            "type": "object",
            "anyOf": [
                {"properties": {"service": {"type": "string"}}, "required": ["service"]},
                {"properties": {"name": {"type": "string"}}, "required": ["name"]},
            ],
            "properties": {"max_per_second": {"type": "integer"}, "sample_rate": {"type": "number"}},
        },
    }
```

Reply:
Ah you're right, I thought that too, but couldn't figure out what the syntax would look like without it. Thanks!